### PR TITLE
Update structure marshaling guidance for .NET Core 3.0

### PR DESCRIPTION
**Pull request by nxtn**
Contributes to #11396 

Continued from #12101

> dotnet/coreclr#22041 fixed marshaling explicit-layout structures on non-Windows platforms.

/cc @jkoritzinsky

### CODE REVIEW

**Comment by NextTurn** [comment]
@mairaw The code owner @jkoritzinsky is not requested for review. #15355

**Review by jkoritzinsky — APPROVED** [review]

**Comment by Youssef1313** [comment]
> 
> 
> @mairaw
> 
> @jkoritzinsky is not identified as the code owner. #15355

https://help.github.com/en/github/creating-cloning-and-archiving-repositories/about-code-owners

> The people you choose as code owners must have write permissions for the repository.

I'm not sure if he has write permissions in dotnet/docs repo, but if he doesn't have write permissions then that is the problem.

**Comment by NextTurn** [comment]
> I'm not sure if he has write permissions

The approval marks of those who have write access are green.

**Comment by Youssef1313** [comment]
> 
> 
> > I'm not sure if he has write permissions
> 
> The approval marks of those who have write access are green.

Well, then this is the problem.

**Comment by mairaw** [comment]
We're revisiting all the assignments, code owners, perms, etc. I'm still in the process of doing this after Ron left but I can review and approve this in the meantime given that @jkoritzinsky already approved this.

File: docs/standard/native-interop/customize-struct-marshaling.md
Comment (mairaw) [inline docs/standard/native-interop/customize-struct-marshaling.md]: I'd just say 3.0 since we already shipped and only preview 7 had a go live license

```suggestion
**❌ AVOID** using `LayoutKind.Explicit` when marshaling structures on non-Windows platforms if you need to target runtimes before .NET Core 3.0. The .NET Core runtime before 3.0 doesn't support passing explicit structures by value to native functions on Intel or AMD 64-bit non-Windows systems. However, the runtime supports passing explicit structures by reference on all platforms.
```

**Review by mairaw — CHANGES_REQUESTED** [review]
Small request before we merge

**Review by mairaw — APPROVED** [review]
Thanks for making the additional change.

**Pull request merged**
